packages/react/src/primitives/types/style.ts

```diff
@@ -5,6 +5,13 @@ import { GridItemStyleProps, GridContainerStyleProps } from './grid';
 import { ImageStyleProps } from './image';
 import { TextAreaStyleProps } from './textArea';
 
+/**
+ * Extract string-like types. Defaulted to string if prop doesn't allow it
+ */
+export type StyleProp<PropertyType> = PropertyType extends string
```

Comment:
This is the important part: this generic type will extract the string-like types only, allowing to keep IDE auto-completion, and defaulting to string if no string-like type is found.

Examples:
- `StyleProp<"a" | "b" | "c"> => "a" | "b" | "c"`
- `StyleProp<"a" | "b" | "c" | number> => "a" | "b" | "c" | string`
- `StyleProp<"a" | "b" | "c" | string> => "a" | "b" | "c" | string`
- `StyleProp<any> => string`
- `StyleProp<unknown> => string`

Reply:
Can't we support `number` in addition to string?

Reply:
Fixed in last commit. Now number-compatible types are: `number | string | ("other" | "string" | "literals")`